test/auth/oauth-with-idtoken-validation.tests.js

```diff
@@ -0,0 +1,376 @@
+              })
+            );
+          }
+        };
+        var oauthWithValidation = new OAUthWithIDTokenValidation(oauth, {
+          clientId: 'foobar',
+          clientSecret: CLIENT_SECRET,
+          domain: 'brucke.auth0.com',
+          supportedAlgorithms: ['RS256']
+        });
+        oauthWithValidation.create(PARAMS, DATA, function(e) {
+          expect(e.message).to.eq('jwt audience invalid. expected: foobar');
+          done();
+        });
+      });
+      it('when `token.iss` is invalid', done => {
+        var oauth = {
+          create: function() {
+            return new Promise(res =>
+              res({
+                id_token:
+                  'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6Ik5FVkJOVU5CT1RneFJrRTVOa1F6UXpjNE9UQkVNRUZGUkRRNU4wUTJRamswUmtRMU1qRkdNUSJ9.eyJuaWNrbmFtZSI6ImpvaG5mb28iLCJuYW1lIjoiam9obmZvb0BnbWFpbC5jb20iLCJwaWN0dXJlIjoiaHR0cHM6Ly9zLmdyYXZhdGFyLmNvbS9hdmF0YXIvMzhmYTAwMjQyM2JkOGM5NDFjNmVkMDU4OGI2MGZmZWQ_cz00ODAmcj1wZyZkPWh0dHBzJTNBJTJGJTJGY2RuLmF1dGgwLmNvbSUyRmF2YXRhcnMlMkZqby5wbmciLCJ1cGRhdGVkX2F0IjoiMjAxOC0wOS0xMlQyMToxMzo1OS42OTNaIiwiZW1haWwiOiJqb2huZm9vQGdtYWlsLmNvbSIsImVtYWlsX3ZlcmlmaWVkIjpmYWxzZSwiaXNzIjoiaHR0cHM6Ly9icnVja2UuYXV0aDAuY29tLyIsInN1YiI6ImF1dGgwfDVhMjA1NGZmNDUxNTc3MTFiZTgxODJmNCIsImF1ZCI6IkVxbjdHTUV3VzhDbmN1S2FhcFRuNWs5VEJ0MzRQdldmIiwiaWF0IjoxNTM2Nzg2ODM5LCJleHAiOjE1MzY4MjI4Mzl9.BkWn4lTu-_GWBC9QfjN1fux4yNe7TfjoyVwsd6tWc7GpuIsAb6GtZbiijfVBkiCchp7V28U2APMTi5Wt1luKkdD0OgI28GOFKKp6qM1qbpt1kexr3So5TgfTb9xbQF-B2HCrqE-fGughAOD4qc4N4UPS_6vRz24fyb4Y8O1wPtdfg9h49ioDa-c3-gyYsaWtUqRJfFoVU9AXBZaIJnKoefz5Oz-_cRWxOI5ci_zWWE2BoxkIMqWVN3Xzzr2njdFKM22HrIIzCz23neNW7bzyMotGjG0B4dbKaY4oIGiK7nI4OopDKbK1AD8KZpTjw97SpL8MbnIhN8-1c1fIVOxKEA'
+              })
+            );
+          }
+        };
+        var oauthWithValidation = new OAUthWithIDTokenValidation(oauth, {
+          clientSecret: CLIENT_SECRET,
+          domain: 'auth.brucke.club',
+          supportedAlgorithms: ['RS256']
+        });
+        oauthWithValidation.create(PARAMS, DATA, function(e, d) {
+          expect(e.message).to.eq('jwt issuer invalid. expected: https://auth.brucke.club/');
```

Comment:
I issue we don't have any control over the error message here? I wonder if it's bad practice to show what we were expecting.